sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -244,6 +250,12 @@ private object DateTimeFormatterHelper {
           for (style <- unsupportedNarrowTextStyle if patternPart.contains(style)) {
             throw new IllegalArgumentException(s"Too many pattern letters: ${style.head}")
           }
+          if (bugInStandAloneForm && (patternPart.contains("LLL") || patternPart.contains("qqq"))) {
```

Comment:
You checked only 'LLL' pattern in `bugInStandAloneForm()` but throws exception for 'qqq' as well. Are you sure they are directly related?

Reply:
In the java doc:
```
Pattern letters 'L', 'c', and 'q' specify the stand-alone form of the text styles.
```

I think they are directly related. And I tested `q` locally as well.

Reply:
ok. I have double checked the pattern 'qqq'. It has the same problem

JDK 11:
```sql
spark-sql> select to_csv(named_struct('date', date '1970-01-01'), map('dateFormat', 'qqq', 'locale', 'RU'));
1-й кв.
spark-sql> select to_csv(named_struct('date', date '1970-01-01'), map('dateFormat', 'qqq', 'locale', 'EN'));
Q1
```

JDK 8
```sql
spark-sql> select to_csv(named_struct('date', date '1970-01-01'), map('dateFormat', 'qqq', 'locale', 'RU'));
1
spark-sql> select to_csv(named_struct('date', date '1970-01-01'), map('dateFormat', 'qqq', 'locale', 'EN'));
1
```